src/csvcubed/utils/sparql_handler/data_cube_inspector.py

```diff
@@ -151,14 +185,33 @@ def get_cube_identifiers_for_data_set(
 
     def get_dsd_qube_components_for_csv(self, csv_url: str) -> QubeComponentsResult:
         """
-        Getter for DSD Qube Components cached property, Raises a KeyError if the csv_url
-        is not associated with a CubeComponentResult.
+        Get DSD Qube Components for the given csv url.
         """
         return get_from_dict_ensure_exists(self._dsd_qube_components, csv_url)
 
     def get_shape_for_csv(self, csv_url: str) -> CubeShape:
         """
-        Returns the shape of the cube stored in a given CSV file,
-         Raises a KeyError if the csv_url is not associated with a cube.
+        Get the cube shape.
         """
         return get_from_dict_ensure_exists(self._cube_shapes, csv_url)
+
+    def get_code_lists_and_cols(self, csv_url: str) -> CodelistsResult:
+        """
+        Get the codelists and columns associated with the given csv url.
+        """
+
+        return self._codelists_and_cols.get(csv_url, CodelistsResult([], 0))
+
+    def get_suppressed_columns_for_csv(self, csv_url: str) -> List[str]:
```

Comment:
I'm not sure this really adds much over just retrieving the columns definitions yourself for the csv and doing some filtering.

Reply:
See previous comment on this function's use (will delete this function if answer is yes)